### PR TITLE
Add information about composer.json

### PR DESCRIPTION
**Pull request by sypets**
Resolves: #252

### CODE REVIEW

File: Documentation/ExtensionArchitecture/ComposerJson/Index.rst [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+=====================
+
+*-- required*
+
+.. note::
+
+   While the file :file:`composer.json` is currently not strictly required, it is considered
+   bad practice not to add one. That is why we classify it as "required".
+
+Including a :file:`composers.json` is strongly recommended for a number of reasons:
```
Comment (pixeldesu) [inline Documentation/ExtensionArchitecture/ComposerJson/Index.rst]: small typo here composer~s~.json

File: Documentation/ExtensionArchitecture/ComposerJson/Index.rst [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+#. Working with Composer in general is strongly recommended for TYPO3.
+
+   If you are not using Composer for your projects yet, see :ref:`t3install:migrate-to-composer`
+   in the "Installation & Upgrade Guide".
+
+
+Minimal composer.json
+=====================
+
+This is a minimal composer.json for a TYPO3 extensions:
```
Comment (pixeldesu) [inline Documentation/ExtensionArchitecture/ComposerJson/Index.rst]: Depending on what amount of extensions we're talking about, this needs to be structured a bit differently.

Either it is 

> This is a minimal composer.json for a TYPO3 extension:

or

> This is a minimal composer.json for TYPO3 extensions:

File: Documentation/ExtensionArchitecture/ComposerJson/Index.rst [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+=====================
+composer.json
+=====================
+
+*-- required*
+
+.. note::
+
+   While the file :file:`composer.json` is currently not strictly required, it is considered
```
Comment (pixeldesu) [inline Documentation/ExtensionArchitecture/ComposerJson/Index.rst]: Why is it not required? This maybe should be expanded upon like

> While the file composer.json is currently not strictly required _for an extension to function properly_,...

**Comment by sypets** [comment]
@pixeldesu Thanks for the review. I pushed another commit to resolve the problems.

File: Documentation/ExtensionArchitecture/ComposerJson/Index.rst [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+           "typo3/cms-core": "^8.7.8"
+       },
+       "extra": {
+           "typo3/cms": {
+               "extension-key": "extension_key"
+           }
+       }
+   }
+
+* **name** (*required*): `<vendor name>/<dashed extension key>` "Dashed extension key" means that every
```
Comment (DanielSiepmann) [inline Documentation/ExtensionArchitecture/ComposerJson/Index.rst]: I would use a definition list instead of a normal list and bold keys.

**Pull request merged**